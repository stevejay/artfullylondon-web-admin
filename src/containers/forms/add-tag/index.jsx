import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { Field, reduxForm } from 'redux-form'
import Form from '_src/components/form'
import FormRow from '_src/components/form/row'
import TextField from '_src/components/text/field'
import FormButtons from '_src/components/form/buttons'
import Divider from '_src/components/divider'
import { ALLOWED_TAG_TYPES } from '_src/constants/tag'
import tagConstraint from '_src/constants/tag-constraint'
import { TAG_EDITOR_FORM_NAME } from '_src/constants/form'
import './index.m.scss'

export const AddTagForm = ({ submitting, handleSubmit }) => (
  <Form styleName='container' onSubmit={handleSubmit}>
    <FormRow>
      <Field
        label='Tag Name'
        name='label'
        component={TextField}
        required
        maxLength={tagConstraint.label.length.maximum}
        autos={false}
        forceSingleLine
        containerStyle={{ flexBasis: 'auto' }}
      />
    </FormRow>
    <Divider />
    <FormButtons submitting={submitting} submitLabel='Add' />
  </Form>
)

AddTagForm.propTypes = {
  initialValues: PropTypes.object.isRequired,
  submitting: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  submitSucceeded: PropTypes.bool.isRequired,
  tagType: PropTypes.oneOf(ALLOWED_TAG_TYPES).isRequired
}

export default connect((_, ownProps) => ({
  initialValues: {
    label: '',
    tagType: ownProps.tagType,
    addTagForEvent: ownProps.addTagForEvent
  }
}))(
  reduxForm({
    form: TAG_EDITOR_FORM_NAME
  })(AddTagForm)
)
