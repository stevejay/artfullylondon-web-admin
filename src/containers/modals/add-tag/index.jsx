import React from 'react'
import PropTypes from 'prop-types'
import ModalContainer from '_src/components/modal/container'
import { ALLOWED_TAG_TYPES } from '_src/constants/tag'
import AddTagForm from '_src/containers/forms/add-tag'

class AddTagModal extends React.Component {
  handleSubmit = values => {
    this.props.onSubmit(values)
  }
  render () {
    const { tagType, title } = this.props

    return (
      <ModalContainer title={title} type='narrow' onHide={this.props.onHide}>
        <AddTagForm
          onSubmit={this.handleSubmit}
          tagType={tagType}
          addTagForEvent
        />
      </ModalContainer>
    )
  }
}

AddTagModal.namepropTypes = {
  onHide: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  tagType: PropTypes.oneOf(ALLOWED_TAG_TYPES).isRequired,
  title: PropTypes.string.isRequired
}

export default AddTagModal
