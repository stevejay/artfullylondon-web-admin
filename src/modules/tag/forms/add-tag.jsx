import React from 'react'
import PropTypes from 'prop-types'
import { Field, reduxForm } from 'redux-form'

import Form from '_src/components/form'
import FormRow from '_src/components/form/row'
import FormError from '_src/components/form/error'
import TextField from '_src/components/text/field'
import ButtonField from '_src/components/button/field'
import * as formConstants from '_src/constants/form'
import * as tagConstants from '_src/modules/tag/constants'

export const AddTagForm = ({
  canAddTag,
  handleSubmit,
  pristine,
  submitting,
  error
}) => (
  <Form onSubmit={handleSubmit}>
    <FormRow>
      <Field
        label='Tag Name'
        name='label'
        component={TextField}
        required
        maxLength={tagConstants.constraint.label.length.maximum}
        autos={false}
        forceSingleLine
        containerStyle={{ flexBasis: 'auto' }}
      />
      <ButtonField
        label='Add'
        type='submit'
        disabled={pristine || !canAddTag}
        submitting={submitting}
      />
    </FormRow>
    <FormError error={error} hideGenericErrorMessages />
  </Form>
)

AddTagForm.propTypes = {
  canAddTag: PropTypes.bool.isRequired,
  initialValues: PropTypes.object.isRequired,
  pristine: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  reset: PropTypes.func.isRequired,
  tagType: PropTypes.oneOf(tagConstants.ALLOWED_TAG_TYPES).isRequired,
  error: PropTypes.any
}

export default reduxForm({
  form: formConstants.TAG_EDITOR_FORM_NAME,
  enableReinitialize: true,
  touchOnBlur: false
})(AddTagForm)
