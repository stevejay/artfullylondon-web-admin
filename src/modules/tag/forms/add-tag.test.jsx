import React from 'react'
import _ from 'lodash'

import { AddTagForm } from './add-tag'

it('should render correctly', () => {
  const wrapper = shallow(
    <AddTagForm
      canAddTag
      initialValues={{ label: 'The Label' }}
      pristine
      submitting={false}
      handleSubmit={_.noop}
      onSubmit={_.noop}
      reset={_.noop}
      tagType='medium'
    />
  )

  expect(wrapper).toMatchSnapshot()
})

it('should render correctly when text has been entered but the tag cannot currently be added', () => {
  const wrapper = shallow(
    <AddTagForm
      canAddTag={false}
      initialValues={{ label: 'The Label' }}
      pristine={false}
      submitting={false}
      handleSubmit={_.noop}
      onSubmit={_.noop}
      reset={_.noop}
      tagType='medium'
    />
  )

  expect(wrapper).toMatchSnapshot()
})
