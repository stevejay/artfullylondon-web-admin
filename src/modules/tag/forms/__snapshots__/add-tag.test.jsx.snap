// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Form
  onSubmit={[Function]}
>
  <FormRow>
    <Field
      autos={false}
      component={[Function]}
      containerStyle={
        Object {
          "flexBasis": "auto",
        }
      }
      forceSingleLine={true}
      label="Tag Name"
      maxLength={50}
      name="label"
      required={true}
    />
    <ButtonField
      disabled={true}
      label="Add"
      submitting={false}
      type="submit"
    />
  </FormRow>
  <FormError
    hideGenericErrorMessages={true}
  />
</Form>
`;

exports[`should render correctly when text has been entered but the tag cannot currently be added 1`] = `
<Form
  onSubmit={[Function]}
>
  <FormRow>
    <Field
      autos={false}
      component={[Function]}
      containerStyle={
        Object {
          "flexBasis": "auto",
        }
      }
      forceSingleLine={true}
      label="Tag Name"
      maxLength={50}
      name="label"
      required={true}
    />
    <ButtonField
      disabled={true}
      label="Add"
      submitting={false}
      type="submit"
    />
  </FormRow>
  <FormError
    hideGenericErrorMessages={true}
  />
</Form>
`;
